portfolio/src/main/webapp/script.js

```diff
@@ -65,3 +65,186 @@ function addRandomFunFact() {
+ * @param {number} numPages The total number of pages. 
+ * @param {number} activePageNumber The previously active page number.
+ * @return {number} The currently active page number.
+ */
+function createPaginationBar(numPages, activePageNumber) {
+  // Set the currently active page number to 1 if the previously active page number no longer exists.
+  if (activePageNumber > numPages) {
+    activePageNumber = 1;
+  }
+  
+  pageListElement = document.getElementById('page-list');
+  pageListElement.innerHTML = '';
+
+  pageListElement.appendChild(createPageElement('Previous'));
+  for (var i = 1; i <= numPages; i++) {
+    if (i == activePageNumber) {
+      pageListElement.appendChild(createPageElement(i, true));
+    } else {
+      pageListElement.appendChild(createPageElement(i, false));
+    }
+  }
+  pageListElement.appendChild(createPageElement('Next'));
+
+  return activePageNumber;
+}
+
+/**
+ * Creates an <li> element for a page item in the pagination bar. 
+ * @param {string} text The innerText of the <a> child element of the <li>.
+ * @param {boolean} active Whether the created page item is active.
+ * @return {!Element<li>} The page element created.
+ */
+function createPageElement(text, active) {
+  const pageElement = document.createElement('li');
+  pageElement.className = 'page-item';
+  if (active) { 
+    pageElement.classList.add('active'); 
+  }
+  
+  const linkElement = document.createElement('a');
+  linkElement.className = 'page-link';
+  linkElement.innerText = text;
+  pageElement.appendChild(linkElement);
+
+  pageElement.addEventListener('click', event => changePage(event));
+  return pageElement;
+}
+
+/**
+ * Changes the active page element and displays the comments on that page when a page element is clicked.
+ * @param {!Event} event The click event that triggers the change of page.
+ */
+function changePage(event) {
+  const currentPage = document.querySelector('.page-item.active');
+  const pageText = event.currentTarget.firstElementChild.innerText;
+  
+  if (pageText == 'Previous') {
+    // If the currentPage is the first page, then return without change.
+    if (currentPage.firstElementChild.innerText == 1) { 
+      return; 
+    }
+
+    currentPage.previousSibling.classList.add('active');
+  } else if (pageText == 'Next') {
+    // If the currentPage is the last page, then return without change.
+    const numPages = document.querySelectorAll('#page-list li').length;
+    if (currentPage.firstElementChild.innerText == numPages - 2) { 
+      return; 
+    }
+
+    currentPage.nextSibling.classList.add('active');
+  } else {
+    event.currentTarget.classList.add('active');
+  }
+
+  currentPage.classList.remove('active');
+
+  // Fetch comments on that page without recreating pagination.
+  fetchComments(false);
+}
+
+/**
+ * Fetches the comments from the data server and displays the comments on the currently active page.
```

Comment:
Just to make it simpler:
```suggestion
 * Fetches the comments from the data server and displays them on the currently active page.
```

Reply:
Got it